src/Components/Shared/SearchBar.tsx

```diff
@@ -0,0 +1,170 @@
+import React from "react"
+import { InputGroup, Dropdown, FormControl, Form } from "react-bootstrap"
+import DropdownButton from "react-bootstrap/DropdownButton"
+import { GetSearchResults } from "../../Endpoints/Search"
+
+import Accordion from "react-bootstrap/Accordion"
+import Card from "react-bootstrap/Card"
+import { ArtistObject } from "../../Models/SpotifyObjects/ArtistObjects"
+import { SimplifiedAlbumObject } from "../../Models/SpotifyObjects/AlbumObjects"
+import { TrackObject } from "../../Models/SpotifyObjects/TrackObjects"
+import { SimplifiedPlaylistObject } from "../../Models/SpotifyObjects/PlaylistObjects"
+import { searchTypes, items, responseItems } from "../../Models/Custom"
+
+interface SearchBarProps {
+    artist?: boolean
+    playlist?: boolean
+    track?: boolean
+    album?: boolean
+    // eslint-disable-next-line no-unused-vars
+    onSearchSelect(item: items): void
+}
+
+interface SearchBarState {
+    // items?: Array<items> | undefined
+    items?: (ArtistObject | SimplifiedAlbumObject | TrackObject | SimplifiedPlaylistObject)[]
+    searchVal: string
+    types: {
+        artist: boolean
+        playlist: boolean
+        track: boolean
+        album: boolean
+    }
+}
+
+export default class SearchBar extends React.Component<SearchBarProps, SearchBarState> {
+    constructor(props: SearchBarProps) {
+        super(props)
+
+        this.state = {
+            searchVal: "",
+            types: {
+                artist: props.artist ?? false,
+                playlist: props.playlist ?? false,
+                track: props.track ?? false,
+                album: props.album ?? false,
+            },
+        }
+
+        this.search = this.search.bind(this)
+        this.onRecordSelect = this.onRecordSelect.bind(this)
+    }
+
+    search(e: React.KeyboardEvent): void {
+        if (e.key !== "Enter") {
+            return
+        }
+
+        GetSearchResults(this.createQuery()).then((res) => {
+            let itms: Array<items> = []
+
+            for (const i in res) {
```

Comment:
This kind of data "shaping" is another thing I think should be in service files.